docs/backhandler.md

```diff
@@ -3,62 +3,166 @@ id: backhandler
+* **If no subscription returns true or none are registered,** it programmatically invokes the default back button functionality to exit the app.
 
-iOS: Not applicable.
+> **Warning for modal users:** If your app shows an opened `Modal`, `BackHandler` will not publish any events ([see `Modal` docs](modal#onrequestclose)).
 
-The event subscriptions are called in reverse order (i.e. last registered subscription first), and if one subscription returns true then subscriptions registered earlier will not be called. Beware: If your app shows an opened `Modal`, BackHandler will not publish any events ([see `Modal` docs](modal#onrequestclose)).
-
-Example:
+## Pattern
 
 ```jsx
 BackHandler.addEventListener('hardwareBackPress', function() {
-  // this.onMainScreen and this.goBack are just examples, you need to use your own implementation here
-  // Typically you would use the navigator here to go to the last state.
+  /**
+   * this.onMainScreen and this.goBack are just examples,
+   * you need to use your own implementation here.
+   *
+   * Typically you would use the navigator here to go to the last state.
+   */
 
   if (!this.onMainScreen()) {
     this.goBack();
+    /**
+     * When true is returned the event will not be bubbled up
+     * & no other back action will execute
+     */
     return true;
   }
+  /**
+   * Returning false will let the event to bubble up & let other event listeners
+   * or the system's default back action to be executed.
+   */
   return false;
 });
 ```
 
-Lifecycle example:
+## Example 
+
+The following example implements a scenario where you confirm if the user wants to exit the app:
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      Function Component Example
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      Class Component Example
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+```SnackPlayer name=BackHandler&supportedPlatforms=android
+import React, { useEffect } from "react";
+import { Text, View, StyleSheet, BackHandler, Alert } from "react-native";
+
+export default function App() {
+  useEffect(() => {
+    const backAction = () => {
+      Alert.alert("Hold on!", "Are you sure you want to go back?", [
+        {
+          text: "Cancel",
+          onPress: () => null,
+          style: "cancel"
+        },
+        { text: "YES", onPress: () => BackHandler.exitApp() }
+      ]);
+      return true;
+    };
+
+    const backHandler = BackHandler.addEventListener(
+      "hardwareBackPress",
+      backAction
+    );
+
+    return () => backHandler.remove();
```

Comment:
Question: should this example use `Backhandler.removeEventListener()`?

Reply:
`removeEventListener` is a static method that requires passing the reference of the event handler to clear the event listener. 
However, `addEventListener` already returns a `NativeEventSubscription`
<img width="766" alt="Screenshot 2020-03-04 at 4 31 12 AM" src="https://user-images.githubusercontent.com/6841445/75828182-47a89500-5dd1-11ea-99fb-ee3e81faaed3.png">
which can be cleared by calling the remove method which feels more straightforward 
<img width="375" alt="Screenshot 2020-03-04 at 4 31 24 AM" src="https://user-images.githubusercontent.com/6841445/75828264-732b7f80-5dd1-11ea-8be9-604e61739dc5.png">

If we need to explain the usage of `removeEventListener` it should be a separate example

Reply:
I feel like that needs covering as well. Ideally the example should include all of the API—or have text explaining where and how to use the bits that aren't covered.

Reply:
Seems like I need to come up with an explanation & an example on this 🤔

Reply:
Would you please, @DaniAkash? Would really complete the docs!

Reply:
Added the explanation!